dot/parachain/provisioner/provisioner.go

```diff
@@ -15,26 +15,39 @@ import (
 
 var logger = log.NewFromGlobal(log.AddContext("pkg", "parachain-provisioner"))
 
-// we expect an inherent to be ready after this time.
-const InherentTimeout = 2 * time.Second
+// inherentPreProposeTimeout is the duration to wait before inherent data is ready
+const inherentPreProposeTimeout = 2 * time.Second
+
+func New() *Provisioner {
+	return &Provisioner{
+		perRelayParent: make(map[common.Hash]perRelayParent),
+
+		// Buffer size 2: one active delay + safety margin
+		availableInherent: make(chan common.Hash, 2),
+	}
+}
 
 type Provisioner struct {
+	perRelayParent map[common.Hash]perRelayParent
+
 	// TODO #4162
```

Comment:
Should this comment be removed? Which part of #4162 does this PR not cover? The acceptance criteria of #4162 mention handling timeouts, so I guess it is that.

Reply:
No, that's a different. issue which is not covered in this PR.